### DISCUSSION TITLE
Deploy on Google Cloud App Engine

### DISCUSSION THREAD

**Discussion by jpedrosouza** [adonisjs/core#2392] [Help]
I've been trying to deploy my app to the Google Cloud App Engine for 3 hours, but I'm having errors with regard to environment variables. I've tried to do everything to solve, but it did not help anything, resort to you is my last alternative, before redoing the whole project without the adonis. I look forward to a little help.

First Error

`Exception: E_MISSING_ENV_VALUE: Missing environment variable "HOST"      at Object.ensureValue (/workspace/node_modules/@adonisjs/env/build/src/Schema/helpers.js:26:15)      at Object.validate [as HOST] (/workspace/node_modules/@adonisjs/env/build/src/Schema/string.js:44:19)      at /workspace/node_modules/@adonisjs/env/build/src/Env/index.js:87:60      at Array.forEach (<anonymous>)      at Env.process (/workspace/node_modules/@adonisjs/env/build/src/Env/index.js:86:44)      at Application.loadEnvironmentVariables (/workspace/node_modules/@adonisjs/application/build/src/Application.js:266:18)      at Application.setup (/workspace/node_modules/@adonisjs/application/build/src/Application.js:467:14)      at HttpServer.wire (/workspace/node_modules/@adonisjs/core/build/src/Ignitor/HttpServer/index.js:48:26)      at HttpServer.start (/workspace/node_modules/@adonisjs/core/build/src/Ignitor/HttpServer/index.js:146:24)      at Object.<anonymous> (/workspace/server.ts:19:37)`

Second Error

`Exception: E_MISSING_ENV_VALUE: Missing environment variable "CACHE_VIEWS"      at Object.ensureValue (/workspace/node_modules/@adonisjs/env/build/src/Schema/helpers.js:26:15)      at Object.validate [as CACHE_VIEWS] (/workspace/node_modules/@adonisjs/env/build/src/Schema/boolean.js:34:19)      at /workspace/node_modules/@adonisjs/env/build/src/Env/index.js:87:60      at Array.forEach (<anonymous>)      at Env.process (/workspace/node_modules/@adonisjs/env/build/src/Env/index.js:86:44)      at Application.loadEnvironmentVariables (/workspace/node_modules/@adonisjs/application/build/src/Application.js:266:18)      at Application.setup (/workspace/node_modules/@adonisjs/application/build/src/Application.js:467:14)      at HttpServer.wire (/workspace/node_modules/@adonisjs/core/build/src/Ignitor/HttpServer/index.js:48:26)      at HttpServer.start (/workspace/node_modules/@adonisjs/core/build/src/Ignitor/HttpServer/index.js:146:24)      at Object.<anonymous> (/workspace/server.ts:19:37)`

**Comment by McSneaky** [comment]
Have you set those 2 env variables?

**Reply by jpedrosouza** [reply]
Yes, I defined them in an .env and copied it to the build folder but it didn't work, I even tried to declare the environment variables in the app.yaml but I was also unsuccessful

**Reply by KeithEinstein** [reply]
Add this to your .adonisrc.json file


"metaFiles": [
".env",
".adonisrc.json",
{
  "pattern": "public/**",
  "reloadServer": false
},
{
  "pattern": "resources/views/**/*.edge",
  "reloadServer": false
}],
